cdap-kubernetes/src/main/java/co/cask/cdap/k8s/common/AbstractWatcherThread.java

```diff
@@ -161,7 +161,7 @@ public final void run() {
 
         // We just retry on any form of exceptions
         Throwable cause = e.getCause();
-        if (cause instanceof IOException || cause instanceof IllegalStateException) {
+        if (cause instanceof IOException || e instanceof IllegalStateException) {
```

Comment:
why is this change needed?

Reply:
Because the K8s API don't wrap IllegalStateException.